Java/TelemetrySystem/src/main/java/tddmicroexercises/telemetrysystem/TelemetryClientImpl.java

```diff
@@ -2,17 +2,27 @@
 
 import java.util.Random;
 
-public class TelemetryClientImpl extends TelementryConnectionImpl implements TelementryClient
-{
+public class TelemetryClientImpl implements TelementryClient {
     public static final String DIAGNOSTIC_MESSAGE = "AT#UD";
 
     private String diagnosticMessageResult;
 
-    private final Random connectionEventsSimulator = telementryConnectionEventSimulator.getConnectionEventsSimulator();
+    private final Random connectionEventsSimulator;
 
-    public void send(String message) {
+    private final TelementryConnectionEventSimulatorImpl telementryConnectionEventSimulator;

```

Comment:
change the type to interface name(TelementryConnectionEventSimulator), so it depends on Abstraction than on concrete class. Hence, loosely Coupled

Reply:
this statement is applicable for any object which has an interface / abstract class